vms/platformvm/txs/fee/calculator_test.go

```diff
@@ -0,0 +1,515 @@
+			unsignedTx: transferSubnetOwnershipTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:      "AddPermissionlessValidatorTx Primary Network pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func(t *testing.T) txs.UnsignedTx {
+				return addPermissionlessValidatorTx(t, constants.PrimaryNetworkID)
+			},
+			expected: feeTestsDefaultCfg.AddPrimaryNetworkValidatorFee,
+		},
+		{
+			name:      "AddPermissionlessValidatorTx Subnet pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func(t *testing.T) txs.UnsignedTx {
+				subnetID := ids.GenerateTestID()
+				require.NotEqual(t, constants.PrimaryNetworkID, subnetID)
+				return addPermissionlessValidatorTx(t, subnetID)
+			},
+			expected: feeTestsDefaultCfg.AddSubnetValidatorFee,
+		},
+		{
+			name:      "AddPermissionlessDelegatorTx Primary Network pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func(t *testing.T) txs.UnsignedTx {
+				return addPermissionlessDelegatorTx(t, constants.PrimaryNetworkID)
+			},
+			expected: feeTestsDefaultCfg.AddPrimaryNetworkDelegatorFee,
+		},
+		{
+			name:      "AddPermissionlessDelegatorTx pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func(t *testing.T) txs.UnsignedTx {
+				subnetID := ids.GenerateTestID()
+				require.NotEqual(t, constants.PrimaryNetworkID, subnetID)
+				return addPermissionlessDelegatorTx(t, subnetID)
+			},
+			expected: feeTestsDefaultCfg.AddSubnetDelegatorFee,
+		},
+		{
+			name:       "BaseTx pre EUpgrade",
+			chainTime:  upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: baseTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:       "ImportTx pre EUpgrade",
+			chainTime:  upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: importTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:       "ExportTx pre EUpgrade",
+			chainTime:  upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: exportTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:      "RewardValidatorTx pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func(_ *testing.T) txs.UnsignedTx {
+				return &txs.RewardValidatorTx{
+					TxID: ids.GenerateTestID(),
+				}
+			},
+			expected: 0,
+		},
+		{
+			name:      "AdvanceTimeTx pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func(_ *testing.T) txs.UnsignedTx {
+				return &txs.AdvanceTimeTx{
+					Time: uint64(time.Now().Unix()),
+				}
+			},
+			expected: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			uTx := tt.unsignedTx(t)
+			fc := NewStaticCalculator(feeTestsDefaultCfg, upgrades)
+			require.Equal(t, tt.expected, fc.CalculateFee(uTx, tt.chainTime))
+		})
+	}
+}
+
+func addValidatorTx(t *testing.T) txs.UnsignedTx {
```

Comment:
All of this tx initialization helper code to me is really confusing... I understand that for static fees this isn't needed and we're putting it in this PR because dynamic fees will have to use some initialized txs since we need to introspect tx data to create fees. I feel like this makes more sense to put in the dynamic fees PR instead of here? I suspect we can get away with something simpler that initializes a lot less

Reply:
Done. Simplified the creators to return empty txs (exceptions are where I need to specify the subnetID) and move the fully formed txs to next PR, https://github.com/ava-labs/avalanchego/pull/2703, where different content will result in diffent fees